src/domain/navigation/index.js

```diff
@@ -14,23 +14,23 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import {Navigation, Segment} from "./Navigation.js";
-import {URLRouter} from "./URLRouter.js";
+import { Navigation, Segment } from "./Navigation.js";
+import { URLRouter } from "./URLRouter.js";
 
 export function createNavigation() {
     return new Navigation(allowsChild);
 }
 
-export function createRouter({history, navigation}) {
-    return new URLRouter({history, navigation, stringifyPath, parseUrlPath});
+export function createRouter({ history, navigation }) {
+    return new URLRouter({ history, navigation, stringifyPath, parseUrlPath });
 }
 
 function allowsChild(parent, child) {
-    const {type} = child;
+    const { type } = child;
     switch (parent?.type) {
         case undefined:
             // allowed root segments
-            return type === "login"  || type === "session" || type === "sso" || type === "logout";
+            return type === "login" || type === "session" || type === "sso" || type === "logout" || type === "oidc-callback" || type === "oidc-error";
```

Comment:
Why not just go with one `oidc` segment for both an error and callback, with different values? Would reduce the amount of boilerplate I think.